### ISSUE TITLE
price deviation - moving back to blockchain data

### ISSUE THREAD

**Issue by ashachaf** [bancorprotocol/webapp#1023]
**Describe the bug**
currently, we pull the API data on site load and never refresh it.
when checking the pool balances we use the information provided in the API.
IF the pool was upgraded, the data we use is stale and point us to a converter with 0 reserve balances.
since API is 12 blocks behind, even if we pull the api every 15sec, this will not help as we would have ~3min of incorrect data.

**solution**
from the API, take the pool token address.
on the blockchain:
- use the token address to identify pool converter
- use pool converter to fetch reserve balances

**Comment by Velua** [comment]
Okay so in the event a **converter is upgraded** where the anchor remains the same but converter address changes, the pools liquidity will move to the new converter but the API will remain referencing the old converter with no liquidity in it therefore give back bad data. 

We'll need to identify which pools need this manual blockchain sourced check rather than using the API. 
Can we rely on the ConverterRegistry which matches Anchors and Converter addresses to detect such mismatches? I imagine the contract registry will reference the latest information?

**Comment by yudilevi** [comment]
The API is actually updated with the right address after update but it takes a few mins.
The best would be to simply query the anchor owner when loading the popup. I imagine we have a few things we need to load from the blockchain when that popup opens, so we can do a multicall to get all the stuff we need in a single request.

In general, pages that lead to user actions (as opposed to "read only" pages like the data page) should rely on blockchain data to minimize dependencies and lower the chance of using stale data.

**Comment by Velua** [comment]
>  The best would be to simply query the anchor owner when loading the popup. I imagine we have a few things we need to load from the blockchain when that popup opens, so we can do a multicall to get all the stuff we need in a single request.

Well we'd be achieving the same thing by just checking the bulk anchors and converters from the converter registry won't we?

**Comment by yudilevi** [comment]
> > The best would be to simply query the anchor owner when loading the popup. I imagine we have a few things we need to load from the blockchain when that popup opens, so we can do a multicall to get all the stuff we need in a single request.
> 
> Well we'd be achieving the same thing by just checking the bulk anchors and converters from the converter registry won't we?

Yeah, for sure, it might just be too expensive - no point in getting all converters when opening the popup since next time the user opens it for another pool, we'll need to get it all again. So might as well just query the one you need.
